### PR TITLE
Multiple chats

### PR DESCRIPTION
**Pull request by fmzbl**
- Update store to keep track of the chat history
- Refactor chat flow to support multiple chat history
- Create chat history widget
- Small fixes and style

### CODE REVIEW

File: moxin-frontend/src/data/chat.rs [hunk 1]
```diff
@@ -24,29 +28,87 @@ impl ChatMessage {
...
+                    truncated.push_str(ellipsis);
+                    truncated
+                } else {
+                    message.content.clone()
+                };
+
+                self.set_title(title);
+            }
+        }
+    }
```
Comment (joulei) [inline moxin-frontend/src/data/chat.rs:111]: I think this is fine for now, we can iterate on this later and ask the model come up with a title based on the context.

File: moxin-frontend/src/data/store.rs [hunk 2]
```diff
@@ -53,19 +55,21 @@ pub struct ModelWithPendingDownloads {
 
 #[derive(Default)]
 pub struct Store {
-    // This is the backend representation, including the sender and receiver ends of the channels to
-    // communicate with the backend thread.
+    /// This is the backend representation, including the sender and receiver ends of the channels to
+    /// communicate with the backend thread.
```
Comment (joulei) [inline moxin-frontend/src/data/store.rs:59]: Thank you for catching these, I've been updating some as I find them

File: moxin-frontend/src/chat/chat_history.rs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+live_design! {
+    import makepad_widgets::base::*;
+    import makepad_widgets::theme_desktop_dark::*;
+
+    import crate::shared::styles::*;
+    import makepad_draw::shader::std::*;
+
+    ChatCard = {{ChatCard}} {
+        content = <RoundedView> {
```
Comment (joulei) [inline moxin-frontend/src/chat/chat_history.rs:15]: Just a tip for the future, if the reason you're creating a `content = <RoundedView>` is that you want `ChatCard` to have the look/behaviour of `RoundedView`, you can do so directly by 'inhering':
`ChatCard = {{ChatCard}}<RoundedVIew> { ... }` in that way you don't have to deal with accessing `self.view(id!(content));`.
I wouldn't bother with changing this now though.

**Review by joulei — APPROVED** [review]
Great job! 
This looks good, thank you for cleaning up names and docs as you go 🫶

File: moxin-frontend/src/chat/chat_history.rs (hunk 3 above)
Comment (fmzbl) [reply moxin-frontend/src/chat/chat_history.rs:15]: thats exactly it yeah, thanks.

**Review by jmbejar — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**